File: py/hey_programmer/hey_programmer_test.py
import unittest

from hey_programmer import greet


class TestHeyProgrammer(unittest.TestCase):
    def test_case_one(self):
        str = "programmer"
        self.assertEqual(greet(str), "hey programmer")

    def test_case_two(self):
        str = "alvin"
        self.assertEqual(greet(str), "hey alvin")

    def test_case_three(self):
        str = "jason"
        self.assertEqual(greet(str), "hey jason")

    def test_case_four(self):
        str = "how now brown cow"
        self.assertEqual(greet(str), "hey how now brown cow")


if __name__ == "__main__":
    unittest.main()

File: go/hey_programmer/hey_programmer_test.go
package hey_programmer

import "testing"

func TestHeyProgrammer(t *testing.T) {
	testCases := []struct {
		name  string
		input string
		want  string
	}{
		{"test case one", "programmer", "hey programmer"},
		{"test case two", "alvin", "hey alvin"},
		{"test case three", "jason", "hey jason"},
		{"test case four", "how now brown cow", "hey how now brown cow"},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			got := hey(tc.input)

			if got != tc.want {
				t.Errorf("got %v, want %v", got, tc.want)
			}
		})
	}
}
